activiti-core-common/activiti-expression-language/src/main/java/org/activiti/core/el/EnvironmentVariableELResolver.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2010-2020 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.core.el;
+
+import jakarta.el.ELContext;
+import jakarta.el.ELResolver;
+
+public class EnvironmentVariableELResolver extends ELResolver {
+
+    private static final String VAR_PREFIX = "vars";
+    public static final String VAR_PREFIX_WITH_DOT = VAR_PREFIX + ".";
+
+    @Override
+    public Object getValue(ELContext context, Object base, Object property) {
+        if (base == null && VAR_PREFIX.equals(property)) {
+            context.setPropertyResolved(true);
+            return new EnvVar();
+        }
+        if (base instanceof EnvVar && property != null) {
+            String env = System.getenv(VAR_PREFIX_WITH_DOT + property);
+            if(env != null) {
```

Comment:
I think it should always set the property as resolved, even when the env is null or not set. Said it differently, I think you can remove this condition.